src/coreclr/vm/eeconfig.cpp

```diff
@@ -901,12 +901,18 @@ fTrackDynamicMethodDebugInfo = CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_
 
         tieredCompilation_CallCountingDelayMs = CLRConfig::GetConfigValue(CLRConfig::INTERNAL_TC_CallCountingDelayMs);
 
+        bool hasSingleProcessor;
 #ifndef TARGET_UNIX
-        bool hadSingleProcessorAtStartup = CPUGroupInfo::HadSingleProcessorAtStartup();
-#else // !TARGET_UNIX
-        bool hadSingleProcessorAtStartup = g_SystemInfo.dwNumberOfProcessors == 1;
-#endif // !TARGET_UNIX
-        if (hadSingleProcessorAtStartup)
+        if (CPUGroupInfo::CanEnableThreadUseAllCpuGroups())
+        {
+            hasSingleProcessor = CPUGroupInfo::GetNumActiveProcessors() == 1;
+        }
+        else
+#endif
+        {
+            hasSingleProcessor = GetCurrentProcessCpuCount() == 1;
```

Comment:
Given that process affinity masks can change, is it possible various calls to GetCurrentProcessCpuCount in the same process could end up returning different values, such that for example ProcessorCount might return > 1 but hasSingleProcessor is true?  Or does GetCurrentProcessCpuCount cache and forever return the same value after that?

Reply:
The value is computed at runtime startup and cached forever.